chemprop/cli/predict.py

```diff
@@ -140,13 +145,9 @@ def add_predict_args(parser: ArgumentParser) -> ArgumentParser:
 
 
 def process_predict_args(args: Namespace) -> Namespace:
-    if args.output is None:
-        args.test_path = Path(args.test_path)
-        name = f"{args.test_path.stem}_preds.csv"
-        args.output = args.test_path.with_name(name)
-    else:
-        args.output = Path(args.output)
-
+    args.output = args.output or Path(args.test_path.stem + "_preds.csv")
+    if args.output.suffix not in [".csv", ".pkl"]:
+        args.output = Path(str(args.output) + ".csv")
```

Comment:
```suggestion
        args.output = args.output.with_suffix(".csv")
```

Reply:
This could be good, and if I make this change, I should make a similar change in convert.py. I think `with_suffix` is clear which is good. There are two reasons I didn't originally use it, both related to preventing user error. What do you think?
1. `-o mypreds.plk` is obviously an typo for mypreds.pkl. If the output is `mypreds.plk.csv`, it would be immediately clear to the user why their predictions weren't saved as a pkl. 
2. If the user doesn't put an extension on their file name, but does use "." in their file name (`data.2021.external`), part of their file name would be removed.

Reply:
> 1. `-o mypreds.plk` is obviously an typo for mypreds.pkl. If the output is `mypreds.plk.csv`, it would be immediately clear to the user why their predictions weren't saved as a pkl.

This will get saved `mypreds.csv`. You can `log.info` this as well for a user to know that we're coercing their output into a CSV:

```
INFO: Specified output path ("mypreds.plk") has unrecognized suffix (".plk")! Changing output path to "mypreds.csv" ...
```

> 2. If the user doesn't put an extension on their file name, but does use "." in their file name (`data.2021.external`), part of their file name would be removed.

IMO it's not on us to ensure some simple default logic catches all the corner cases so long as we document what it does. We can always `log.info` this stuff as well so users aren't caught totally by surprise, but it seems overly cumbersome to implement logic for all of these corner cases. And FWIW, an input of `data.2021.external` would already be coerced into `data.2021.csv` with the current logic (the new logic would actually write it to `data.csv`). Both of those seem equally bad to me, but dots in file extensions are commonly understood to be file extension separators rather than an alternative to dashes or underscores.